pkg/skaffold/runner/build_deploy.go

```diff
@@ -73,18 +73,20 @@ func (r *SkaffoldRunner) BuildAndTest(ctx context.Context, out io.Writer, artifa
 			return nil, err
 		}
 
-		if !r.runCtx.SkipTests() {
-			if err = r.tester.Test(ctx, out, bRes); err != nil {
-				return nil, err
-			}
-		}
-
 		return bRes, nil
 	})
 	if err != nil {
 		return nil, err
 	}
 
+	if !r.runCtx.SkipTests() {
+		if err = r.tester.Test(ctx, out, bRes); err != nil {
+			return nil, err
+		}
+	} else {
+		logrus.Warnf("Skip tests flag (skip-tests) is set to %t. Tests won't run.", (r.runCtx.SkipTests()))
```

Comment:
`r.runCtx.SkipTests()` is always true here.

```suggestion
		logrus.Warnln("Skip tests flag (skip-tests) is set to true. Tests won't run.")
```